### DISCUSSION TITLE
Unable to from/import a .NET 8 library

### DISCUSSION THREAD

**Discussion by nickna** [pythonnet/pythonnet#2421] [Q&A] [answered]
I'm trying to pull my .NET library into Python and getting a failure on the from/import statement.

Repro Steps:
1. Git clone [this repo](https://github.com/nickna/Neighborly) (.NET 8 class library)
2. Compile the Neighborly library (defaults to .NET 8 - Any CPU)
3. Write a Python script with Pythonnet installed via pip:
```
clr.AddReference(r"C:\MyPathTo\Neighborly.dll")
from Neighborly import *
```
Error:
```
Traceback (most recent call last):
  File "c:\Code\SampleLib\Test.py", line 18, in <module>
    from Neighborly import *
ModuleNotFoundError: No module named 'Neighborly'
```

Looking at the IL metadata, I can confirm that Neighborly is the assembly name and also the namespace.
![image](https://github.com/user-attachments/assets/0b5b6469-fb69-490c-ab5c-865746c7d841)

I created a sample class with an Add function that compiled and was successfully imported into Python. 
I also added all of the same Nuget dependencies and recompiled them. It continues to call the Add function.
This rules out a dependency load failure.

I'm running on Windows 11 with Python 3.10.11.

Any ideas?  🙏

**Comment by filmor** [comment, marked as answer]
As documented, by default we load a .NET Framework runtime at startup. The simplest option is to run `import pythonnet; pythonnet.load("coreclr")` before importing `clr`.

**Reply by virusestoadventures182** [reply]
doesnt work for me

**Comment by swatikd06** [comment]
I am also facing same error- ModuleNotFoundError: No module named-<for the dll added as reference>
even after adding b:elow lines
import pythonnet; 
pythonnet.load("coreclr") before importing clr.

No change in the error
